### PR TITLE
Issue #270: LineWrappingHandler keeps same indentation level for syntactically parallel elements

### LINKED ISSUE #270: IndentationCheck: keep the same indentation level for syntactically parallel elements
There are cases with the indentation for <a href="http://stackoverflow.com/questions/25611099/what-is-syntactically-parallel-elements-in-java">syntactically parallel elements</a> which should be handled properly by the check.

```
In general, two continuation lines use the same indentation level 
if and only if they begin with syntactically parallel elements.
```

In general, syntactically parallel elements are <a href="http://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.11">separators and operators</a> which are at the same level of nesting. The level of nesting can be figured out by the opening/closing parenthesis.

E.g.:

a) each line is started with "+" as syntactically parallel elements of the same scope

```
foo("bar"
    + "baz"
    + "boo");
```

b) each line is started with "+" but they are not syntactically parallel in same scope( see "(" , ")")

```
foo("bar"
    + foo2("bar"
           + "baz"));
```

There is a more complicated case with builder pattern usage:

```
private Something getSomething() {
    return feedBuilder("data", "/path", "ONE", "TWO").addFiles(
            zipBuilder("file1", "file2")
                    .setUpdateFileSchedule(Schedule.longWork(new LocalTime(1, 0)))
                    .addFiles(
                            coolBuilder("otherFile1").build(),
                            coolBuilder("otherFile2").build(),
                            coolBuilder("otherFile3").build(),
                            coolBuilder("otherFile4").build()
                    ).build(),
            zipBuilder("resource1", "resource2")
                    .setUpdateFileSchedule(Schedule.work(new LocalTime(11, 0)))
                    .addFile(coolBuilder("data").build())
                    .build()
            ).build();
}
```

where the first addFiles(...) method receives two parameters (the two zipBuilders) which separated by syntactically parallel elements ",". Each zipBuilder has its own parallel elements (calls of the setUpdateFileSchedule and addFiles methods) and so forth.

We also have to investigate other cases with parallel elements and put them into test inputs.

**NOTE:** the Indentation check's violations mustn't be conflicted with the current <a href="https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheck.java">SeparatorWrapCheck</a> and <a href="https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java">OperatorWrapCheck</a> checks.
##

### PR DESCRIPTION
**Pull request by plan3d**
Resolves #270: LineWrappingHandler keeps same indentation level for syntactically parallel elements, counting expression starts (e.g. nodes with type LPAREN without corresponding RPAREN node on the examined path)

Diff Regression projects: https://gist.githubusercontent.com/plan3d/ca91d8262d18c4c84f6b62fa1cb4bebc/raw/1439fe419e2c073c551ac421ac4767ecb0bb3a0a/projects-to-test-on.properties
Diff Regression config: https://gist.githubusercontent.com/plan3d/ca91d8262d18c4c84f6b62fa1cb4bebc/raw/b14b001997ef1f3cb0f0cf161b8f59041caf5322/my_check.xml

### CODE REVIEW

**Comment by plan3d** [comment]
Dear admins, can you please think about approving issue #270?

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectForAndParameter.java [hunk 1]
```diff
@@ -21,30 +21,30 @@ void fooMethodWithIf() { //indent:2 exp:2
           getString(1000000000, "Loooooooooooooooong")); i < 10; i++) {} //indent:10 exp:>=8
 
     for (Map.Entry<String, String> entry : new SecondForLongName("Loo" //indent:4 exp:4
-        + "ooooooooooooooooooooong").getMap(new //indent:8 exp:8
-        FooForClass(), 10000000, //indent:8 exp:8
-        getString(10000, "Loooooooooo" //indent:8 exp:8
-        + "ooooong")).entrySet()) {} //indent:8 exp:8
+            + "ooooooooooooooooooooong").getMap(new //indent:12 exp:12
```
Comment (Abhishek-kumar09) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectForAndParameter.java:24]: Why this change?, I guess it is following the lineWrapping Indentation.

**Comment by Abhishek-kumar09** [comment]
It's a good check, but still lots of projects don't follow it strictly, and therefore lots of new errors are seen in projects.

Can we alter it somehow like, if `forceStrictCondition` is true, then only we will throw up these errors.
This check is related to `lineWrappingIndentation` and `forceStrictCondition` is meant to be used with `lineWrappingIndentation`.

**Comment by plan3d** [comment]
Good idea, I will change the check to throw errors only if `forceStrictCondition` is true.

**Comment by romani** [comment]
PR is still in draft mode, CI full of problems.

@plan3d , in few month we might close draft PR to keep on leash PRs that we have. You can continue working on issue at your own pace without PR.
PR is for state when you need feedback/help or your are ready.

**Comment by plan3d** [comment]
I will close this pull request and open a new one when code is ready for review
